Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/AnalyserActivity.java

```diff
@@ -0,0 +1,125 @@
+import com.example.android.camera2.slowmo.fragments.CameraFragment;
+
+/** This class handles the lifecycle of lag calculation. */
+public class AnalyserActivity extends AppCompatActivity {
+  private final Integer FILE_PICKER_REQUEST_CODE = 10;
+
+  private TextView filePath;
+  private TextView analyseResultField;
+  private Button filePickerBtn;
+  private Button analyzeBtn;
+  private Intent fileIntent;
+  private Uri fileUri;
+
+  private VideoProcessor videoProcessor;
+  private LagCalculator lagCalculator;
+  private ResultPublisher resultPublisher;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_analyser);
+
+    filePath = findViewById(R.id.selectedFilePath);
+    filePickerBtn = findViewById(R.id.fileSelectButton);
+    analyzeBtn = findViewById(R.id.analyze);
+    analyseResultField = findViewById(R.id.analyseResultView);
+
+    analyseResultField.setMovementMethod(new ScrollingMovementMethod());
+
+    videoProcessor = new VideoProcessor();
+
+    lagCalculator = new LagCalculator(videoProcessor, CameraActivity.serverHandler);
+
+    resultPublisher = new ResultPublisher();
+
+    analyseResultField.append(
+        "\nPort bound:" + CameraActivity.serverHandler.getServerSocketPort() + "\n");
+
+    CameraFragment.Companion.getFilePath();
+    filePath.setText(CameraFragment.Companion.getFilePath());
+
+    filePickerBtn.setOnClickListener(
+        new View.OnClickListener() {
+          @Override
+          public void onClick(View view) {
+            fileIntent = new Intent(Intent.ACTION_GET_CONTENT);
+            fileIntent.setType("*/*");
+            startActivityForResult(fileIntent, FILE_PICKER_REQUEST_CODE);
+          }
+        });
+
+    analyzeBtn.setOnClickListener(
+        new View.OnClickListener() {
+          @RequiresApi(api = Build.VERSION_CODES.P)
+          @Override
+          public void onClick(View view) {
+            try {
+              analyze();
+            } catch (Exception e) {
+              Log.d(ContentValues.TAG, "Analyse call failed with: " + e.getMessage());
+            }
+          }
+        });
+  }
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+    if (requestCode == FILE_PICKER_REQUEST_CODE && resultCode == RESULT_OK) {
+      String path = data.getData().toString();
+      fileUri = data.getData();
+      filePath.setText(path);
+    }
+  }
+
+  /** Triggers subsequent lifecycle events of lag calculations. */
+  @RequiresApi(api = Build.VERSION_CODES.P)
+  private void analyze() {
+    fileUri = Uri.parse(filePath.getText().toString());
+    videoProcessor.setVideoReader(getApplicationContext(), fileUri);
+
+    CameraActivity.serverHandler.downloadServerTimeStamps();
+
+    videoProcessor.doOcrOnVideo(lagCalculator);
```

Comment:
Just doOcr ?  Also should it be doOCR? not sure what the Java style here is.

Reply:
done